lib/reporters/html.js

```diff
@@ -313,6 +313,27 @@ function error(msg) {
   document.body.appendChild(fragment('<div id="mocha-error">%s</div>', msg));
 }
 
+var policy = {
+  createHTML: function(html) {
+    /**
+     * Note that this policy lets html through unchanged. This is potentially
+     * a security vulnerability if untrusted data is set to innerHTML, as it
+     * allows arbitrary code execution.
+     *
+     * Ideally this code would be refactored to not use .innerHTML, and this
+     * policy deleted, or this policy could return the html after it has been
+     * processed by a secure sanitization system like dompurify
+     */
+    return html;
+  }
+};
+if (
```

Comment:
fwiw, `window` will not be `undefined`; the `html` reporter will throw mad exceptions without `window`.

also, you can use `const` or `let` instead of `var` across the codebase now

Reply:
+1, switched these changes to use `let`

Dropped the `window` check here. Should I drop it from highlight-tags.js as well? If someone is importing highlight-tags but not calling it from node that could fail, because we're eagerly initializing the policy.

Reply:
Looks like the window check is needed, tests fail without it: https://github.com/mochajs/mocha/pull/4447/checks?check_run_id=1171116538

I think the html reporter is sometimes imported from a node environment even if it isn't used.

Reply:
yeah, that's a bug.  ok. ty